### PR TITLE
fix: add new alerts columns

### PR DESCRIPTION
**Pull request by rebelchris**
Added new alerts columns as described in [DR](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/659161109/Referral+program+and+profile+UX+improvements#Persistent-data-for-referral-reminder).
had to adjust from Date as it's not supported in JSONB, decided to go for bigint to avoid y2k issues down the line (2038)

WT-1921 #done

### CODE REVIEW

File: src/routes/boot.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export type Experimentation = {
 
 export type BaseBoot = {
   visit: { visitId: string; sessionId: string };
-  alerts: Omit<Alerts, 'userId'>;
+  alerts: Omit<Alerts, 'userId' | 'flags'>;
```
Comment (idoshamun) [inline src/routes/boot.ts:68]: We have to make sure we don't select this column also

Comment (rebelchris) [reply src/routes/boot.ts:68]: Lee made it a [seperate](https://dailydotdev.atlassian.net/jira/software/projects/WT/boards/11?selectedIssue=WT-1924) ticket, if thats ok with you?

Comment (idoshamun) [reply src/routes/boot.ts:68]: The ticket refers to the mutation to update alerts. I talk about the query that fetches alerts from the database for boot. If we introduce it in a different ticket it means there will be a period of time that users get this information even though they shouldn't

Comment (rebelchris) [reply src/routes/boot.ts:68]: @idoshamun I added a function for now, let me know what you think.

**Review by idoshamun — APPROVED** [review]
Awesome!

**Comment by rebelchris** [comment]
@idoshamun @capJavert Added the function now as a generic one we can reuse.

Queries should happen like this:
```
SELECT * FROM alerts WHERE dateCastIndex(flags, 'lastReferralReminder') >= '2020-04-20'
```

And we can easily add new ones:
```
CREATE INDEX on alerts ((dateCastIndex(flags, 'lastReferralReminder')))
```

Result using the index fn:
Screenshot 2023-11-02 at 12 56 48

**Review by capJavert — APPROVED** [review]
Looks great 🚀

**Review by idoshamun — APPROVED** [review]

**Pull request merged**